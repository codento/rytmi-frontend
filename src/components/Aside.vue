<template>
  <aside class="aside-menu" />
</template>

<script>
export default {
  name: 'c-aside'
}
</script>
