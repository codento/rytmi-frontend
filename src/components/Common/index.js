import ProjectRow from './ProjectRow.vue'
import SkillRow from './SkillRow.vue'
import SkillExplanations from './SkillExplanations.vue'
import UtilizationChart from './UtilizationChart.vue'
import CollapsableItem from './CollapsableItem.vue'
import EducationCard from './EducationCard.vue'
import ProjectForm from './ProjectForm.vue'
import ProjectProfileForm from './ProjectProfileForm.vue'

export {
  ProjectRow,
  SkillRow,
  SkillExplanations,
  UtilizationChart,
  CollapsableItem,
  EducationCard,
  ProjectForm,
  ProjectProfileForm
}
