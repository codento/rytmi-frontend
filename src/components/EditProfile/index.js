import EditLinks from './EditLinks.vue'
import EditSkills from './EditSkills.vue'
import ProfileForm from './ProfileForm.vue'
import EditProjects from './EditProjects.vue'
import WorkHistoryEntryEdit from './WorkHistoryEntryEdit.vue'
import WorkHistoryEntryList from './WorkHistoryEntryList.vue'
import CvInfo from './CvInfo.vue'
import CvInfoEditIntroduction from './CvInfoEditIntroduction.vue'
import CvInfoEditEducation from './CvInfoEditEducation.vue'
import CvInfoEditEducationForm from './CvInfoEditEducationForm.vue'

export {
  EditLinks,
  EditSkills,
  ProfileForm,
  EditProjects,
  WorkHistoryEntryList,
  CvInfo,
  CvInfoEditIntroduction,
  CvInfoEditEducation,
  CvInfoEditEducationForm,
  WorkHistoryEntryEdit
}
