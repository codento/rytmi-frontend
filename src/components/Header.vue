<template>
  <header class="app-header navbar">
    <button class="navbar-toggler mobile-sidebar-toggler d-lg-none" type="button" @click="mobileSidebarToggle">
      <span class="navbar-toggler-icon"></span>
    </button>
    <b-link class="navbar-brand" to="Home"></b-link>
    <button class="navbar-toggler sidebar-toggler d-md-down-none" type="button" @click="sidebarToggle">
      <span class="navbar-toggler-icon"></span>
    </button>
     <b-navbar-nav  v-if="isAuthenticated">
      <b-nav-item to="/search">Search</b-nav-item>

      <b-nav-item v-if="!getProfileId" to="/createProfile">Create Profile</b-nav-item>
      <b-nav-item v-else to="/editProfile">Edit My Profile</b-nav-item>

    </b-navbar-nav>
    <b-navbar-nav class="ml-auto">
      <b-nav-item id="loginBtn"  v-if="!isAuthenticated" v-on:click="login" >Sign in</b-nav-item>
      <b-nav-item id="logoutBtn" v-if="isAuthenticated" v-on:click="logout" >Sign out</b-nav-item>
      <b-nav-item id="logoutBtn"  v-on:click="checkStatus" >Check status</b-nav-item>
    </b-navbar-nav>
    <button class="navbar-toggler aside-menu-toggler d-md-down-none" type="button" @click="asideToggle">
      <span class="navbar-toggler-icon"></span>
    </button>
    
  </header>
</template>
<script>
import {mapGetters, mapActions} from 'vuex'
import {loadAuthClient, handleLogin, handleLogout} from '../utils/auth'

export default {
  name: 'c-header',
  computed: {
    ...mapGetters(['isAuthenticated', 'getProfileId', 'getToken'])
  },
  methods: {
    sidebarToggle (e) {
      e.preventDefault()
      document.body.classList.toggle('sidebar-hidden')
    },
    sidebarMinimize (e) {
      e.preventDefault()
      document.body.classList.toggle('sidebar-minimized')
    },
    mobileSidebarToggle (e) {
      e.preventDefault()
      document.body.classList.toggle('sidebar-mobile-show')
    },
    asideToggle (e) {
      e.preventDefault()
      document.body.classList.toggle('aside-menu-hidden')
    },
    checkStatus () {
      this.$toasted.global.rytmi_success({
        message: 'Authenticated: ' + this.isAuthenticated
      })
    },
    logout () {
      handleLogout().then(() => {
        this.logoutAuth()
        this.$router.push(this.$route.query.redirect || '/home')
      })
    },
    login () {
      handleLogin().then((response) => {
        this.requestAuth(response.Zi.id_token)
      }).catch((error) => {
        console.log(error)
        this.$toasted.global.rytmi_error({
          message: 'Login failed'
        })
      })
    },
    ...mapActions(['requestAuth', 'logoutAuth', 'createUser'])
  },
  created () {
    /* eslint-disable */
    loadAuthClient()
  }
}
</script>
