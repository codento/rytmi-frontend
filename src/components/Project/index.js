import ProjectForm from './ProjectForm.vue'
import ProjectList from './ProjectList.vue'
import ProjectProfileForm from './ProjectProfileForm.vue'

export {
  ProjectForm,
  ProjectList,
  ProjectProfileForm
}
