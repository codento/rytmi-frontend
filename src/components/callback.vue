<template>
  
</template>

<script>
import {mapActions} from 'vuex'

export default {
  created () {
    this.$nextTick(() => {
      this.fetchProfiles()
      this.fetchSkills()
      this.fetchProfileSkills()
      this.$router.push(this.$route.query.redirect || '/')
    })
  },
  methods: {
    ...mapActions([
      'fetchProfiles',
      'fetchSkills',
      'fetchProfileSkills'
    ])
  }
}
</script>