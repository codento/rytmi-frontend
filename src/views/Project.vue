<template>
  <div class="animated fadeIn profile-editor">
      <h1>{{project.name}}</h1>
      <p>{{project.description}}</p>
      <p>Start: {{new Date(project.startDate).toLocaleString()}}</p>
      <p>End: {{new Date(project.endDate).toLocaleString()}}</p>
      <h1>Members</h1>
      <div v-for="profile in projectProfilesByProjectId(project.id)" :key="profile.id">
        {{profileById(profile.id)}}aaa
      </div>
      <ProjectProfileForm :projectId="project.id" />
  </div>
</template>

<script>
import { mapGetters, mapActions } from 'vuex'
import {
  ProjectProfileForm
} from '../components/Project'

export default {
  name: 'Project',
  data: {
    project: {}
  },
  computed: {
    ...mapGetters([
      'projectById',
      'projectProfilesByProjectId',
      'profileById'
    ]),
    ...mapActions([
      'fetchProjectProfiles'
    ]),
    project () {
      return this.projectById(this.$route.params.id)
    }
  },
  components: {
    ProjectProfileForm
  },
  mounted () {
    this.$store.dispatch('fetchProjectProfiles', this.project)
  }
}
</script>

<style scoped />