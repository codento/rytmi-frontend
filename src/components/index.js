import Aside from './appStructures/Aside.vue'
import Breadcrumb from './appStructures/Breadcrumb.vue'
import Callout from './helpers/Callout.vue'
import Footer from './appStructures/Footer.vue'
import Header from './appStructures/Header.vue'
import Sidebar from './appStructures/Sidebar/Sidebar.vue'

export {
  Aside,
  Breadcrumb,
  Callout,
  Footer,
  Header,
  Sidebar
}
