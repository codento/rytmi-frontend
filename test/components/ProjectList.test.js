import { mount, createLocalVue } from 'vue-test-utils'
import { ProjectList } from '../../src/components/Project'
import Vuex from 'vuex'

const localVue = createLocalVue()

localVue.use(Vuex)

const getters = {
  projects: () => (arg) => arg
}
const store = new Vuex.Store({ getters })

describe('ProjectList.test.js', () => {
  it('Template is correct', () => {
    let wrapper = mount(ProjectList, {
      store,
      localVue
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
