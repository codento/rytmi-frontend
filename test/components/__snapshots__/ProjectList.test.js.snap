// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectList.test.js Template is correct 1`] = `"<b-container><h2>Projects</h2> </b-container>"`;
